draft-ietf-quic-transport.md

```diff
@@ -3874,145 +3874,136 @@ The details of loss detection and congestion control are described in
-devices, each endpoint independently validates ECN counts and disables ECN if
-the path is not showing consistent support for ECN.
+An ECN count is not incremented when no QUIC packets from the received IP
+packet are processed. A QUIC packet detected by a receiver as a duplicate also
+does not increase an ECN count; see {{security-ecn}} for relevant security
+concerns.
 
-Endpoints validate ECN for packets sent on each network path independently.  An
-endpoint thus validates ECN on new connection establishment, when switching to a
-server's preferred address, and on active connection migration to a new path.
-{{ecn-alg}} describes one possible algorithm for testing paths for ECN support.
 
-Even if an endpoint does not use ECN markings on packets it transmits, the
-endpoint MUST provide feedback about ECN markings received from the peer if they
-are accessible.  Failing to report ECN counts will cause the peer to disable ECN
-marking.
+### ECN Validation {#ecn-validation}
 
+It is possible for faulty network devices to corrupt or erroneously drop
+packets that carry a non-zero ECN codepoint. To ensure connectivity in the
+presence of such devices, an endpoint validates the ECN counts for each network
+path and disables use of ECN on that path if errors are detected.
 
-#### Sending ECN Markings
+To perform ECN validation for a new path:
 
-To start ECN validation, an endpoint SHOULD do the following when sending
-packets on a new path to a peer:
+* The endpoint sets an ECT(0) codepoint in the IP header of early outgoing
+  packets sent on a new path to the peer ({{!RFC8311}}).
 
-* Set the ECT(0) codepoint in the IP header of early outgoing packets sent on a
-  new path to the peer ({{!RFC8311}}).
+* The endpoint monitors whether all packets sent with an ECT codepoint are
+  eventually deemed lost (Section 6 of {{QUIC-RECOVERY}}), indicating
+  that ECN validation has failed.
 
-* If all packets that were sent with the ECT(0) codepoint are eventually deemed
-  lost (Section 6 of {{QUIC-RECOVERY}}), validation is deemed to have failed.
+If an endpoint has cause to expect that IP packets with an ECT codepoint might
+be dropped by faulty network element, the endpoint could set an ECT codepoint
+for only the first ten outgoing packets on a path, or for a period of three
+RTTs. If all packets marked with non-zero ECN codepoints are subsequently lost,
+it can disable marking on the assumption that the marking causes in loss.
 
-To reduce the chances of misinterpreting congestive loss as packets dropped by a
-faulty network element, an endpoint could set the ECT(0) codepoint for only the
-first ten outgoing packets on a path, or for a period of three RTTs, whichever
-occurs first.
+An endpoint thus attempts to use ECN and validates this for each new connection,
+when switching to a server's preferred address, and on active connection
+migration to a new path.  {{ecn-alg}} describes one possible algorithm.
```

Comment:
```suggestion
An endpoint negotiates ECN for each new connection or new path.
{{ecn-alg}} describes one possible algorithm.
```

Reply:
I don't want to say negotiates.  And I'm not sure that we need the extreme economy you propose otherwise.

Reply:
I also think the original is much clearer.